common/kvstore/src/main/java/org/apache/spark/util/kvstore/LevelDBIterator.java

```diff
@@ -182,23 +193,34 @@ public boolean skip(long n) {
 
   @Override
   public synchronized void close() throws IOException {
-    db.notifyIteratorClosed(this);
+    db.notifyIteratorClosed(it);
     if (!closed) {
-      it.close();
-      closed = true;
-      next = null;
+      try {
+        it.close();
+      } finally {
+        closed = true;
+        next = null;
+        cancelResourceClean();
```

Comment:
Yes, we have discussed this issue. The reason for not directly calling `this.cleaner.clean()` is because the close process in `Cleaner` has added the operation of `synchronized (this._db)`, which is slightly different from the semantics of the original `close()` method. For specific discussions, please refer to this thread: 

https://github.com/apache/spark/pull/43502#discussion_r1372954706

Reply:
Heh, forgot my own comment :-)